samples/demo-contract-framework/src/main/java/org/eclipse/dataspaceconnector/demo/contract/offer/ContractDescriptor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ *  Copyright (c) 2021 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+package org.eclipse.dataspaceconnector.demo.contract.offer;
+
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Describes a policy-based contract offered by the system.
+ *
+ * The {@link AssetSelectorExpression} defines which assets this contract applies to. Access control policy defines the non-public requirements for accessing a set of assets
+ * governed by the contract. These requirements are therefore not advertised to the agent. For example, access control policy may require an agent to be in a business partner tier.
+ * Usage policy defines the requirements governing use an agent must follow when accessing the data. This policy is advertised to agents as part of a contract.
+ *
+ * A participant agent may access a contract (and its associated assets) if the union of the access policy and usage policy is satisfied by the agent.
+ */
+public class ContractDescriptor {
+    private String id;
+    private AssetSelectorExpression selectorExpression;
+    private Policy accessControlPolicy;
+    private Policy usagePolicy;
```

Comment:
I like that those types are modeled separately.
General observation: as accessControl is not known to the caller the visibility of a ContractOffer may change between calls (e.g. a caller gets downgraded form PARTNER = Gold to Silver). Not sure if that would also affect actual Contracts? Or are access policies just taken into account during contract offer + negotiation phase, but later just the existence of contract counts?

Reply:
Good question and catch. We need to take into account the case where a participant has access to an asset and then the accessControlPolicy is changed, causing them to no longer have access. If there is an outstanding contract and in force, the connector will likely need to honor it. Let's discuss this and then based on that, file an issue as this will affect multiple work streams.

Reply:
As contracts needs to be immutable I'd store actual contracts in a separate store with a unique Id. If we don't do it like this the number of assets or policies may change afterwards, which I think would be a problem. Also a stored contract doesn't need an access policy.

If we do it like this maybe rename the `ContractDiscriptor` to `ContractOfferDescriptor`.

Reply:
rename usagePolicy to contractPolicy

Reply:
Created issue for separate PR